### DISCUSSION TITLE
Nested lists could appear better

### DISCUSSION THREAD

**Discussion by boxallar** [TechbeeAT/jtxBoard#941] [Ideas]
when I write a journal in markdown, nested lists do not display well.
They are rendered with the same bullet points and the indentation is very slight.  At least on my phone, with the font/font sizes I use it is difficult to see the nesting at all.

**Comment by patrickunterwegs** [comment]
Hi @boxallar , Markdown is actually done in this library: 
https://github.com/ArnyminerZ/MarkdownText

@ArnyminerZ , do you maybe have an idea on how to get this solved?

**Reply by boxallar** [reply]
I see. It says that nested lists are not supported as well as other things:

-     Placeholders are not supported.
-     Blockquotes are not supported.
-     Nested lists are not supported.
-     Code blocks are not supported.
-     Tables are not supported.
-     Horizontal rules do not fill the whole width.

Maybe this will change in the future, or jtx will move to another library for markdown?

**Reply by ArnyminerZ** [reply]
It might have a solution. MarkdownText works with the official [Jetbrains Markdown library](https://github.com/JetBrains/markdown), so all the features there should be supported. However, there are more problems on rendering the formatted values than parsing them.

As discussed before, it's preferred to keep using the Compose's Text component, instead of using different components for everything. This causes the "Horizontal rules do not fill the whole width." limitation, since there's no way of actually drawing a line that covers the whole width, at least, one that I know of.

The rest of not supported features are more rendering issues.

Tables are a complicated topic, and it's not easy to make them scroll horizontally, for example, so they must use the same width as the Text component, which is not really good for drawing large ones, only smaller 2-3 short columns.

I would have to look more into nested lists, it should be not that hard to implement them, however, right now I don't have much spare time, so if anybody wants to help, the library is [available at Github](https://github.com/ArnyminerZ/MarkdownText).

Code blocks are not supported simply because I wanted to differenciate them from code spans (which are supported). However, if we don't support specific language syntax highlighting we can just display them like the code spans.

Placeholders is a limitation of the Jetbrains library, at least at the moment I developed the original library, maybe they are now supported, I will have to look into it.

**Comment by rogercreagh** [comment]
> Code blocks are not supported simply because I wanted to differenciate them from code spans (which are supported). However, if we don't support specific language syntax highlighting we can just display them like the code spans.

Currently the text is hidden in a code block when rendered - this feels like a bug rather than a design choice.  If code blocks (or any element) are not supported then the content should be rendered as plain text either with or without the backticks showing.

Blockquotes are also not supported but in that case the text is still rendered and the opening `>` is hidden. Personally I would expect the `>` and the ` ``` ` symbols to be rendered as plain text if they are not supported so that you can see what's going on.

It would be good if blockqoutes could also be supported.
